cmd/kubeadm/app/cmd/join.go

```diff
@@ -145,6 +145,11 @@ func (j *Join) Run(out io.Writer) error {
 		return err
 	}
 
+	err = kubenode.CheckForNodeNameDuplicates(connectionDetails)
```

Comment:
@luxas it should be before tls bootstrap, otherwise it will try to issue new cert for same node name, so looks at right place.